src/rabbit_networking.erl

```diff
@@ -389,13 +389,29 @@ force_connection_event_refresh(Ref) ->
     [rabbit_reader:force_event_refresh(C, Ref) || C <- connections()],
     ok.
 
-handshake(Ref, ProxyProtocol) ->
-    case ProxyProtocol of
+failed_to_recv_proxy_header(Ref, Error) ->
+    Msg = case Error of
+        closed -> "error when receiving proxy header: TCP socket was ~p prematurely";
+        _Other -> "error when receiving proxy header: ~p"
+    end,
+    rabbit_log:error(Msg, [Error]),
+    % The following call will clean up resources then exit
+    _ = ranch:handshake(Ref),
+    exit({shutdown, failed_to_recv_proxy_header}).
```

Comment:
@michaelklishin I don't think that this line will be reached in a real `{error, closed}` situation as `ranch:handshake` should exit [here](https://github.com/ninenines/ranch/blob/master/src/ranch.erl#L251-L253). But, there's no harm in leaving this `exit` call in.